File: leetcode_learning/code_cpp/26.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int size = nums.size();
        if(size==0)
            return 0;
        int i = 0;
        for(int j=1;j<size;j++)
        {
            if(nums[j]!=nums[i])
            {
                i++;
                nums[i] = nums[j];
            }
        }
        return i+1;
        
        //如果要求返回前i个元素的数组，可以添加下面的代码
        // for(int index = i+1;index<size;index++)
        // {
        //     nums.pop_back();
        // }
        // return nums;
    }
};

File: leetcode_learning/auto-leetcode/26.删除有序数组中的重复项/26.py
"""
26.删除有序数组中的重复项
难度：简单
链接：https://leetcode.cn/problems/remove-duplicates-from-sorted-array/
"""



class Solution:
    def removeDuplicates(self, nums: list[int]) -> int:
        if not nums:
            return 0

        slow = 0  # 慢指针：指向当前不重复元素的末尾位置

        for fast in range(1, len(nums)):
            # 发现新元素（与慢指针指向的元素不同）
            if nums[fast] != nums[slow]:
                slow += 1  # 慢指针前进一步
                nums[slow] = nums[fast]  # 将新元素放到正确位置

        return slow + 1  # 不重复元素的个数




